hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java

```diff
@@ -1030,6 +1035,18 @@ public static Map<String, List<Map<String, Expression<?>>>> partitionStrategyCan
         return result;
     }
 
+    /**
+     * Check should be used during the optimized IMapByKey plan construction.
+     * It throws {@link QueryException} of query is analyzed for optimized IMapByKey plans.
+     * @param isAnalyzed is query analyzed
+     */
+    static void checkIMapByKeyPlanIsAnalyzed(boolean isAnalyzed) {
+        if (isAnalyzed) {
+            throw QueryException.error("ANALYZE statement is not applicable for key-based optimized IMap access plans." +
```

Comment:
I would work on a better message. The user is not required to know what plan _will be chosen by the optimizer_ and the SQL syntax is ordinary. Something like "This query uses key-based optimized IMap access plan. ANALYZE is unable to produce meaningful execution statistics for it."

Reply:
https://github.com/hazelcast/hazelcast/pull/26002/commits